MTM-65302 Documenting host restriction in microservice manifest (#4008)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2025-400-0-restring-host-in-probes.md

---
date: 
title: Changing probes setting in microservice manifests 
product_area: Platform services
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-OG_650_b2
    label: Core platform
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-65302
version: 2025.400.0
---

Previously, the host of a liveness or readiness probe of a microservice could be set in the microservice manifest. 

> To avoid possible security implications, we have disabled this possibility. 

Reviewer: ```suggestion
To avoid possible security implications, this option has been disabled. 
```

Reviewer 2: `possibility` sounds better for me, `option` is used in multiple contexts in our system, although here it should be clear from the context.

Author: This part has changed, therefore resolving.

> This improvement does not affect setting the path in the liveness and readiness probe of the microservice which is typically used.

Reviewer 3: which probe or microservice is meant by this sentence?

Author: I have removed the sentence, as it is not really related.

> This improvement does not affect setting the path in the liveness and readiness probe of the microservice which is typically used.

Reviewer: ```suggestion
This change does not affect setting the path in the liveness and readiness probe of the microservice that is typically used.
```

---
date: 
title: Changing probes setting in microservice manifests 
product_area: Platform services
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-OG_650_b2
    label: Core platform
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-65302
version: 2026.23.0
---
Until now, the host of a liveness or readiness probe of a microservice could be set in the microservice manifest. 

> To avoid potential security implications, setting the host in manifest probes will be disabled in Q3 2026 for CD versions and in the 2027 annual release. Uploading a microservice that uses the 'host' parameter within the readiness or liveness probe will be rejected.

Reviewer: ```suggestion
To avoid potential security implications, setting the host in manifest probes will be disabled in Q3 2026 for CD versions and in the 2027 annual release. Uploading a microservice that uses the `host` parameter within the readiness or liveness probe will be rejected.
```

> For testing purposes, the new behavior can already be enabled via a feature toggle 'core.ms-hosting.no-host-in-probes-in-manifest'.

Reviewer: ```suggestion
For testing purposes, the new behavior can already be enabled via a feature toggle `core.ms-hosting.no-host-in-probes-in-manifest`.
```

---
date: 
title: Changing probes setting in microservice manifests 
product_area: Platform services
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-OG_650_b2
    label: Core platform
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-65302
version: 2026.23.0
---

> Previously, the host of a liveness or readiness probe of a microservice could be set in the microservice manifest. 

Reviewer 4: ```suggestion
Until now, the host of a liveness or readiness probe of a microservice could be set in the microservice manifest. 
```

> To avoid possible security implications, setting the host in the manifest probes will be disabled in the future.

Reviewer 4: ```suggestion
To avoid potential security implications, setting the host in manifest probes will be disabled in Q3 2026 for CD versions and in the 2027 annual release. Uploading a microservice that uses the 'host' parameter within the readiness or liveness probe will be rejected.
```

> The removal is planned for Q2 2026 in CD versions and in the 2027 annual release.

Reviewer 4: ```suggestion
```

> The introduced change is currently disabled by default and can be enabled via a feature toggle `core.ms-hosting.no-host-in-probes-in-manifest`.

Reviewer 4: ```suggestion
For testing purposes, the new behavior can already be enabled via a feature toggle 'core.ms-hosting.no-host-in-probes-in-manifest'.
```

{{< c8y-admon-important >}}

> In Q2 2026 for the CD versions and in 2027 for the yearly releases, this change will become **mandatory**.

Reviewer 4: ```suggestion
In Q3 2026 for the CD versions and in 2027 for the yearly releases, the new behavior will become **mandatory**.
```
